cajita/src/Cajita_BovWriter.hpp

```diff
@@ -171,29 +211,45 @@ void writeTimeStep( const int time_step_index, const double time,
                 owned_extents[d] = owned_index_space.extent( d ) + 1;
         }
     }
-    owned_extents[3] = array.layout()->dofsPerEntity();
+    owned_extents[num_space_dim] = array.layout()->dofsPerEntity();
+
+    // Gather halo data if any dimensions are periodic.
+    for ( std::size_t d = 0; d < num_space_dim; ++d )
+    {
+        if ( global_grid.isPeriodic( d ) )
+        {
+            auto halo =
+                createHalo( NodeHaloPattern<num_space_dim>(), 0, array );
+            halo->gather( execution_space(), array );
+            break;
```

Comment:
because we only want to gather 1 time - if any dimension is periodic this achieves the goal of the gather. if none of the dimensions are periodic then we avoid the gather.